common/src/main/scala/explore/components/graphql/SubscriptionRenderMod.scala

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.components.graphql
+
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.vdom.html_<^._
+import react.common._
+import react.semanticui.elements.icon.Icon
+import react.semanticui.sizes._
+import clue.GraphQLStreamingClient
+import cats.implicits._
+import cats.effect.IO
+import crystal._
+import crystal.react.io.implicits._
+import diode.data._
+import diode.react.ReactPot._
+import cats.effect.ContextShift
+import crystal.react.StreamRendererMod
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import cats.effect.Timer
+
+final case class SubscriptionRenderMod[D, A](
+  subscribe:      IO[GraphQLStreamingClient[IO]#Subscription[D]],
+  streamModifier: fs2.Stream[IO, D] => fs2.Stream[IO, A] = identity[fs2.Stream[IO, D]] _
+)(
+  val valueRender: (A, StreamRendererMod.ModState[A]) => VdomNode,
+  val onNewData:   IO[Unit] = IO.unit
+)(
+  implicit val cs: ContextShift[IO],
```

Comment:
I'm wondering about these being `implicit` but I guess it makes constructing these easier?

Reply:
Yes, that's the idea.